### PR TITLE
Added escape for Sphinx characters to default value

### LINKED ISSUE #15: parser with `[*]` in any string causes sphinx warning for `Inline emphasis start-string without end-string`
If I use the following example program:
```python
"""
Test script

.. argparse::
    :module: test
    :func: test_parser
    :prog: test.py
"""

import argparse


def test_parser():
    parser = argparse.ArgumentParser(description='Test script with [*] in description',
                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument('--arg1', help='Argument with [*] in help')
    parser.add_argument('--arg2', help='Another argument with bracketed asterisk in metavar', metavar='[*.blah]')
    return parser


def main():
    parser = test_parser()
    args = parser.parse_args()

    print(args)


if __name__ == '__main__':
    main()
```

It produces the following `Inline emphasis` warning when compiling the sphinx docs.

```
...
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in _build/html.
None:1: (WARNING/2) Inline emphasis start-string without end-string.

Build finished. The HTML pages are in _build/html.

Open in web-browser (chrome/chromium-browser preferred):
	file:///home/user/project/doc/_build/html/index.html
```

It compiles fine and views fine in the output file, but the warnings are being flagged in the pre-commit hooks for my project.

It seems specific to including a bracket with an asterisk in it. I believe the sphinx-argparse parser might need to sanitize the help / description statements before writing out the html. I'm assuming this would also be the case for the epilog as well.

I'm using python 3.8.5 with sphinx 4.0.2 and sphinx-argparse 0.3.1.

### PR DESCRIPTION
**Pull request by RobertoRoos**
Helps to fix #15

### CODE REVIEW

File: test/test_default_html.py [hunk 1]
```diff
@@ -29,6 +29,14 @@
                 (".//section[@id='positional-arguments']", ''),
                 (".//section[@id='positional-arguments']/dl/dt[1]/kbd", 'baz'),
             ],
+            'special-characters.html': [
+                (".//h1", 'Sample', False),
+                (".//h1", 'Special Characters'),
+                (".//section/dl/dd/p", 'Default:'),
+                (".//section/dl/dd/p/code/span", '420'),
+                (".//section/dl/dd/p/code/span", "'*.rst"),
+                (".//section/dl/dd/p/code/span", r"\['\*.rst',"),
```
Comment (ashb) [inline test/test_default_html.py:38]: What does this look like in the rendered HTML (both HTML source, and in a browser) please?

Comment (RobertoRoos) [reply test/test_default_html.py:38]: The result from the test method:
image

In HTML (formatting mine):
```
<p>
Default:
<code class="docutils literal notranslate">
<span class="pre">'*.rst</span> 
<span class="pre">_txt_</span> 
<span class="pre">**strong**</span> 
<span class="pre">*italic*</span> 
<span class="pre">\`\`code\`\`'</span>
</code>
</p>
```

File: sphinxarg/ext.py [hunk 2]
```diff
@@ -142,10 +142,9 @@ def print_action_groups(data, nested_content, markdown_help=False, settings=None
                     '"==SUPPRESS=="',
                     '==SUPPRESS==',
                 ]:
-                    if entry['default'] == '':
-                        arg.append('Default: ""')
-                    else:
-                        arg.append(f"Default: {entry['default']}")
+                    # Put the default value in a literal block - but escape backticks already in the string
+                    default_str = str(entry['default']).replace('`', r'\`')
```
Comment (RobertoRoos) [inline sphinxarg/ext.py:146]: Small problem: the 
```
\`
```
is displayed in the webbrowser literally. I haven't been able to property escape it in Sphinx to display it properly in a literal block.

Comment (RobertoRoos) [reply sphinxarg/ext.py:146]: Hm, actually in plain Sphinx this works fine. The following is rendered as expected:
```
Demo code: ``hi``

Demo code: `````

Demo code: ``\```

Demo code: ```hi```

Demo code: ````hi````

Demo code: ``meta: ``hi````
```

image

**Review by ashb — APPROVED** [review]

**Pull request merged**